### PR TITLE
Remove autocomplete complete from GCSE Maths grade form

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

In [Design spike: GCSE grade options](https://trello.com/c/RjrKbnr0/), we revised how we are using auto selects for GCSE grades. Maths still need to have autocomplete removed from the grade form.

## Changes proposed in this pull request

- Remove the autocomplete
- Add extra validation checks now that grade is entered manually
- Fix broken specs
- 🏖️ 

## Guidance for review
Fill out a GCSE Maths qualification (`/candidate/application/gcse/maths/grade`) and note the lack of _auto-completeness_ when you reach the grade form.

## Link to Trello card

https://trello.com/c/r829S9jB/2558-remove-autocomplete-for-maths-gcse-grades-and-use-the-same-approach-as-that-used-for-science-gcse-grades

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/forms/candidate_interface/gcse_qualification_details_form.rb [hunk 1]
```diff
@@ -72,13 +72,19 @@ def award_year_is_a_valid_date
...
+      else
+        qualification_rexp = invalid_grades[qualification.qualification_type.to_sym]
 
-      if qualification_rexp && grade.match(qualification_rexp)
-        errors.add(:grade, :invalid)
+        errors.add(:grade, :invalid) if qualification_rexp && grade.match(qualification_rexp)
       end
     end
 
+    def sanitize(grade)
```
Comment (george-bit) [inline app/forms/candidate_interface/gcse_qualification_details_form.rb:84]: Hey @TobyRet, just had a go at this in the review app. I submitted a lowercase grade with a space next to it successfully.
image
image
is this intended behaviour?

Should the sanitize method also convert the grade as well as help validate it?

**Review by george-bit — COMMENTED** [review]
LGTM, just want to check RE the sanitize comment

File: app/forms/candidate_interface/gcse_qualification_details_form.rb (hunk 1 above)
Comment (TobyRet) [reply app/forms/candidate_interface/gcse_qualification_details_form.rb:84]: Woops I forgot to save the sanitized grade. I'll fix

Comment (TobyRet) [reply app/forms/candidate_interface/gcse_qualification_details_form.rb:84]: @george-bit - that's all fixed now

https://github.com/DFE-Digital/apply-for-teacher-training/pull/3457/files#diff-821b5429d131177a8a261fc366f7bf33db627f1ac83ae68f4ba9a89ba654d74eR125

Spec added as well

**Review by george-bit — APPROVED** [review]
LGTM :)

**Pull request merged**